server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1435,23 +1439,34 @@ private void closeConsistencyCheck() {
     }
   }
 
-  private void compareToDataInMemory(TabletMetadata tabletMetadata) {
-    if (!tabletMetadata.getFilesMap().equals(getDatafileManager().getDatafileSizes())) {
-      String msg = "Data files in " + extent + " differ from in-memory data "
-          + tabletMetadata.getFilesMap() + " " + getDatafileManager().getDatafileSizes();
-      log.error(msg);
-    }
-  }
-
-  public synchronized void compareTabletInfo(Long updateCounter, TabletMetadata tabletMetadata) {
+  public synchronized void compareTabletInfo(MetadataUpdateCount updateCounter,
+      TabletMetadata tabletMetadata) {
```

Comment:
Does it matter which thread (the main thread vs during a close) is doing the check? This was one thing I was wondering while doing testing. If we add a boolean here of who is calling the check, we could bail if it is not the correct thread.

Reply:
It should not matter which thread is doing the check. The main thing that matters is the order in which the two parameters are that are passed to the method are acquired.  The updateCounter must be acquired before tabletMetadata by the code calling this method (and they must go with the same tablet).

Reply:
Maybe we could add a key extent to the MetadataUpdateCount to allow more validations (make sure counts are for expected extent) in the method.

Reply:
I added some extra validations in e2580d9.